runtime/jcl/common/mgmtruntime.c

```diff
@@ -80,3 +80,24 @@ Java_com_ibm_lang_management_internal_ExtendedRuntimeMXBeanImpl_getProcessIDImpl
 	pid =  (jlong) j9sysinfo_get_pid();
 	return pid;
 }
+
+/**
+ * @return runtime state of the JVM
+ */
+jstring JNICALL
+Java_com_ibm_lang_management_internal_ExtendedRuntimeMXBeanImpl_getVMStateImpl(JNIEnv *env, jclass clazz)
+{
+        J9JavaVM *javaVM = ((J9VMThread *) env)->javaVM;
+	char *state = NULL;
+	switch (javaVM->vmRuntimeStateListener.vmRuntimeState) {
+	case J9VM_RUNTIME_STATE_ACTIVE:
+		state = "Active";
+		break;
+	case J9VM_RUNTIME_STATE_IDLE:
+		state = "Idle";
+		break;
+	default:
+		state = "Unknown";
```

Comment:
This should be an assert - we don't expect the code to ever be anything other than active or idle